mattermost/mattermost-plugin-mscalendar
# [MM-188] Add feature to update custom status and status of the user during meeting

#### Summary

- Added feature to update custom status of the user during meeting
- Added feature to set status of the user from the options provided in the dropdown during the meeting

#### Ticket Link
#228 

#### Screenshots
![image](https://github.com/mattermost/mattermost-plugin-mscalendar/assets/100013900/6ce23597-426f-4f44-bb26-2dfa1152c1a6)
![image](https://github.com/mattermost/mattermost-plugin-mscalendar/assets/100013900/3b4ea791-7480-4a99-9f7f-bd54d7c396aa)


#### What to test?
- Custom status gets updated during a meeting.
- User status gets updated to the selected value during a meeting.
- No change in user status during overlapping meetings or when there are no attendees.
- No change of user custom status during overlapping meetings, when there are no attendees, or when there is already a custom status set.

#### How to test?
- Create a meeting in the MS Calendar.
- Run slash command `/mscalendar settings`, select an option from the `Update Status` setting, and check the updated status on the Mattermost.
- Run slash command `/mscalendar settings`, set the `Set Custom Status` setting to yes, and check the updated custom status on the Mattermost.
